sync/src/block/extension.rs

```diff
@@ -267,28 +267,36 @@ impl NetworkExtension for Extension {
                 }
             }
             SYNC_EXPIRE_TOKEN_BEGIN...SYNC_EXPIRE_TOKEN_END => {
-                let mut requests = self.requests.write();
-                let mut tokens_info = self.tokens_info.write();
-                let token_info = tokens_info.get_mut(&token).unwrap();
-                if token_info.request_id.is_none() {
-                    return
-                }
+                let (id, request_id) = {
```

Comment:
It seems that it's not just a refactoring, but changing the behavior.
Is it safe to not lock `requests` while checking `tokens_info`?

Reply:
It is not refactoring, it's my mistake. And, yes, it is safe.

Reply:
There was no reason to lock `tokens_info` after locking `requests`, and the key point of this patch is to make `token_info.request_id.is_none()` as a flag between `on_timeout()` and `on_peer_response()`.

Reply:
Okay.
@Kais-DkM how about you?

Reply:
It seems to be safe at a first glance. I'll accept it.